Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1056,9 +1074,14 @@ void RenderBlockFlow::layoutBlockChild(RenderBox& child, MarginInfo& marginInfo,
 
     // We are no longer at the top of the block if we encounter a non-empty child.  
     // This has to be done after checking for clear, so that margins can be reset if a clear occurred.
-    if (marginInfo.atBeforeSideOfBlock() && !child.isSelfCollapsingBlock())
+    if (marginInfo.atBeforeSideOfBlock() && !child.isSelfCollapsingBlock()) {
         marginInfo.setAtBeforeSideOfBlock(false);
 
+        if (auto* layoutState = frame().view()->layoutContext().layoutState(); layoutState->peekBlockStartTrimming().has_value()) {
```

Comment:
null deref when no layoutState

Reply:
nit: operator bool calls has_value()

Reply:
Changed both (along with the other calls to `has_value()` )